test/IntegrationSpec.js

```diff
@@ -273,6 +273,36 @@ describe('Integration tests using the demo app', function() {
                 done();
             });
         });
+
+        it('should shard to next level into json response', function(done) {
+            client.get('/shard?type=nextLevelJson',
+                       function(err, req, res, data) {
+                assert.ifError(err);
+                assert.isObject(data);
+                assert.equal(data.name, 'json');
+                done();
+            });
+        });
+
+        it('should shard to next level into json response', function(done) {
+            client.get('/shard?type=nextLevelNotSameLevel',
+                       function(err, req, res, data) {
+                assert.ifError(err);
+                assert.isObject(data);
+                assert.equal(data.name, 'json');
```

Comment:
The next level is reached because it changes to `json`, it was set to `preshard` in handler block 9.